lib/search-suggestions/index.jsx

```diff
@@ -0,0 +1,128 @@
+import React, { Component } from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import SmallSearchIcon from '../icons/search-small';
+import TagIcon from '../icons/tag';
+
+export class SearchSuggestions extends Component {
+  static displayName = 'SearchSuggestions';
+
+  static propTypes = {
+    onSearch: PropTypes.func.isRequired,
+    query: PropTypes.string.isRequired,
+    filteredTags: PropTypes.array.isRequired,
+    storeKeyHandler: PropTypes.func.isRequired,
+  };
+
+  state = {
+    selectedItem: 0,
+  };
+
+  componentDidMount() {
+    this.props.storeKeyHandler({
+      next: this.nextItem,
+      prev: this.prevItem,
+      select: this.selectItem,
+    });
+  }
+
+  componentDidUpdate() {
+    // scroll selected item into view
+    var selectedItem = this.refs.selectedItem;
```

Comment:
check out the [docs for `createRef()`](https://reactjs.org/docs/react-api.html#reactcreateref) to see how we can eliminate some of the string processing here. I updated my understanding of `forwardRef` as well and sadly I don't see doing what we want here. We could get access to an underlying DOM node but it doesn't resolve passing functions up to the parent.

```js
componentDidMount() {
	this.selectedItem= createRef();
}

componentDidUpdate() {
	if ( this.selectedItem.current ) {
		this.selectedItem.current.scrollIntoView(false);
	}
}

render() {
	…

	ref={ selectedItem === index + 1 ? this.selectedItem : null }

	…
}
```

I _think_ this will work and you can see that it removes some manual accounting that we're doing.